src/init.lua

```diff
@@ -234,9 +234,9 @@ function Octree:ChangeNodePosition(node: Node, position: Vector3)
 	if newRegion == node.Region then
 		return
 	end
+	table.insert(newRegion.Nodes, node)
```

Comment:
This was the fix. The new region needs to contain the node before calling `RemoveNode` to avoid an issue where the new region is seen as empty and thus removed from the Octree system. This resulted in nodes being left in regions that had no direct parent to the Octree.